### ISSUE TITLE
-ojunit output xml file format doesnot have any info regarding test source file name

### ISSUE THREAD

**Issue by bkjoshi76** [cpputest/cpputest#903]
usecase : integrating cpputest output xml ( junit) format to the C++ sonar community plugin

c++ sonar community needs to have test src file name for displaying test execution status on the sonar dashboard.

Any option available to get test src  file info in xml file (-ojunit) format?

**Comment by arstrube** [comment]
There is -k `packagename` -- might that help?

**Comment by itavero** [comment]
I do not believe that the JUnit / xUnit XML format allows you to include such data.
See:
* https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd
* http://reflex.gforge.inria.fr/xunit.html#xunitReport

If this data is needed, I think it will need to be in a new output format (so not in `-ojunit`).

**Comment by itavero** [comment]
I also need information about the location of test cases within the sources, however I need to retrieve it *without* running the tests.
See issue #967.

**Comment by itavero** [comment]
This XSD does seem to include the attributes you are looking for:
https://github.com/jenkinsci/xunit-plugin/blob/master/src/main/resources/org/jenkinsci/plugins/xunit/types/model/xsd/junit-10.xsd

**Comment by itavero** [comment]
@bkjoshi76 While working on a solution for #967, I've added a `file` and `line` attribute to each testcase that indicates the position of each test (first line).
I'd like to know if this information will be enough for you to get a better integration within Sonar, or do you need more information to be included in the XML?

My code will currently provide you with an XML file similar to this example:
```xml
<?xml version="1.0" encoding="UTF-8" ?>
<testsuite errors="0" failures="0" hostname="localhost" name="groupName" tests="2" time="0.046" timestamp="2016-05-20T08:07:42">
<properties>
</properties>
<testcase classname="groupName" name="firstTestName" assertions="4" time="0.031" file="/path/to/testsource.cpp" line="23">
</testcase>
<testcase classname="groupName" name="secondTestName" assertions="6" time="0.015" file="/path/to/testsource.cpp" line="34">
</testcase>
<system-out></system-out>
<system-err></system-err>
</testsuite>
```

**Comment by arstrube** [comment]
@itavero I like that the number of assertions is in there. I think it is an important metric - I have seen "tests" with NO assertions :-/.....

**Comment by bkjoshi76** [comment]
Hi,

This is nice solution what i am looking.

I think its good for sonar integration

Regards

Bhaskar joshi

**Comment by itavero** [comment]
@basvodde @bkjoshi76 This one can be closed too, now that PR #1010 has been merged.
